### ISSUE TITLE
[docs] Elastic Reference Docs should include example requests replies.

### ISSUE THREAD

**Issue by jystervinou** [elastic/elasticsearch#26296]
**Describe the feature**:

Some of the Elastic reference docs are, i think, too dense on textual explanations, but too light on examples.

It's easier to understand a given request syntax with examples.

Specifically, the examples are missing the replies. You get the request json, but not the reply.

**My feature request is to add, as much as possible, the *replies* of the requests example.**

Because in [a lot of](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html) [examples](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html) (all examples on these pages, not only the first one in the page) you get the queries, but you don't get the example replies, so you're not sure what you're supposed to get actually.

You could standardize on a simple dataset for all docs (easily accessible in one click), so one's can understand the API, now that she sees the dataset, the request, and the reply. 

To sum it up, the reference docs would then become more tutorial-ish than reference only.

I think that would help understand Elastic's API more quickly.

Thank's for listening.

**Comment by jimczi** [comment]
Thanks @jystervinou , we've started to test our snippets in the documentation some time ago and added a lot of examples where requests and answers are present. For some APIs the results are the same but I agree that we could have more examples of answers in the docs. Are you willing to contribute ? You could start with the example you linked and add an example replies for the `match` and `multi_match` query ?

**Comment by aclowkey** [comment]
I think it's a great idea.  Sometimes users want care about the request, and sometimes about the response, and sometimes both.

I'm not too familiar with asciidocs, is there a way to have a side-by-side view of sources? or tabbed view for 'response' and 'request' ?  or do we have to put them one after the other ?

**Comment by javanna** [comment]
hi @aclowkey I don't think there is a way to have request and response side by side, rather one after the other like we already have in our current docs. Looking forward to your contributions on this.

**Comment by debadair** [comment]
[docs issue triage]

**Comment by jrodewig** [comment]
Except where the response is a simple acknowledgement, response bodies and their parameters are now a standard part of our API documentation, per the API reference template:

https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc

While there are still quite a few snippets missing responses, I feel this addresses the core issue. We'll continue adding missing responses as our ongoing efforts to improve the docs.

If you'd like to see responses documented for any particular APIs, please open a separate issue. Thanks!
